pep-0572.rst

```diff
@@ -359,6 +359,125 @@ found in assignment statements:
+
+site
+^^^^
+
++-----------------------------------------------------+------------------------------------------------------------+
+| Current code                                        | Replaced with                                              |
++-----------------------------------------------------+------------------------------------------------------------+
+|                                                     |                                                            |
+| ::                                                  | ::                                                         |
+|                                                     |                                                            |
+|  env_base = os.environ.get("PYTHONUSERBASE", None)  |   if (env_base := os.environ.get("PYTHONUSERBASE", None)): |
+|  if env_base:                                       |       return env_base                                      |
+|      return env_base                                |                                                            |
+|                                                     |                                                            |
++-----------------------------------------------------+------------------------------------------------------------+
+
+*env_base* is only used on these lines, putting its assignment on the if
+moves it as the "header" of the block.
+
+decimal
+^^^^^^^
++----------------------------------------------+-----------------------------------------------------------------------+
+| Current code                                 | Replaced with                                                         |
++----------------------------------------------+-----------------------------------------------------------------------+
+|                                              |                                                                       |
+| ::                                           | ::                                                                    |
+|                                              |                                                                       |
+|  if self._is_special:                        |   if self._is_special and (ans := self._check_nans(context=context)): |
+|      ans = self._check_nans(context=context) |       return ans                                                      |
+|      if ans:                                 |                                                                       |
+|          return ans                          |                                                                       |
+|                                              |                                                                       |
++----------------------------------------------+-----------------------------------------------------------------------+
+
+Avoid nested if and remove one indentation level.
+
+copy
+^^^^
+
```

Comment:
Maybe add a note here that this example is duplicated from Appendix A -- I don't mind the duplication but I don't want people to wonder if there perhaps is a subtle difference. (Subtlety in use of assignment expressions is an anti-pattern.)